### ISSUE TITLE
🐛 SequelizeOptimisticLockError Exception is Thrown when Executing an External Task

### ISSUE THREAD

**Issue by robpal94** [process-engine/process_engine_runtime#343]
## Description

When an external Task is executed, the ProcessEngine sometimes throws a `SequelizeOptimisticLockError` error. (This error is also propagated to the ExternalTaskWorker, which interprets it as a 500 error). 

The full StackTrace looks like this: 

<details>

```
2019-05-28T10:45:57.422Z - info: [http_extension:error_handler] 500 { SequelizeOptimisticLockError: Attempting to update a stale model instance: ExternalTask
    at constructor.QueryInterface.(anonymous function).then (/Users/robin/Documents/process_engine/process_engine_runtime/node_modules/sequelize/lib/model.js:3817:21)
    at tryCatcher (/Users/robin/Documents/process_engine/process_engine_runtime/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/Users/robin/Documents/process_engine/process_engine_runtime/node_modules/bluebird/js/release/promise.js:517:31)
    at Promise._settlePromise (/Users/robin/Documents/process_engine/process_engine_runtime/node_modules/bluebird/js/release/promise.js:574:18)
    at Promise._settlePromise0 (/Users/robin/Documents/process_engine/process_engine_runtime/node_modules/bluebird/js/release/promise.js:619:10)
    at Promise._settlePromises (/Users/robin/Documents/process_engine/process_engine_runtime/node_modules/bluebird/js/release/promise.js:699:18)
    at _drainQueueStep (/Users/robin/Documents/process_engine/process_engine_runtime/node_modules/bluebird/js/release/async.js:138:12)
    at _drainQueue (/Users/robin/Documents/process_engine/process_engine_runtime/node_modules/bluebird/js/release/async.js:131:9)
    at Async._drainQueues (/Users/robin/Documents/process_engine/process_engine_runtime/node_modules/bluebird/js/release/async.js:147:5)
    at Immediate.Async.drainQueues (/Users/robin/Documents/process_engine/process_engine_runtime/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:794:20)
    at tryOnImmediate (timers.js:752:5)
    at processImmediate [as _immediateCallback] (timers.js:729:5)
  name: 'SequelizeOptimisticLockError',
  modelName: 'ExternalTask',
  values:
   { workerId: '22456552-15f7-4ed8-9b41-becae0ec1bcc',
     lockExpirationTime: 2019-05-28T10:46:27.413Z,
     updatedAt: 2019-05-28T10:45:57.416Z,
     version: 1 },
  where: { id: 98, version: 0 } }
``` 

</details>

## Reproduction

This issue is kinda hard to reproduce, since it seems to happen more or less sporadically.

## My setup

- Node (`node --version`): 8.11.1

**Comment by ElRaptorus** [comment]
Could you describe the Setup in which you encounter this problem? 

Maybe with anonymized BPMNs, a code sample of the worker(s) used, etc. 

Otherwise there's not much we can do.

**Comment by ElRaptorus** [comment]
After doing some digging and conferring with @robpal94 I think I may have an Idea what's going on.

The problem will most likely be with the ExternalTaskWorker:

https://github.com/process-engine/external_task_api_client/blob/develop/typescript/src/external_task_worker.ts#L47

The worker periodically calls `extendLock` on the ExternalTasks that were first polled. 
That is also most likely the request that crashes, judging by the Stacktrace. 

The problem is, the interval that causes the locks to periodically extend never checks if any of the tasks have been finished in the meantime. 
In fact, it may even be possible to run an additional `fetchAndLock` while an `extendLock` request **and** a `finish` request are still active.

So basically, there are multiple requests against the database that run simultaneously that all use the same value for `version`. But with optimistic locking active, this will result in an error for all but the first request.

This weird looping behavior is not very robust anyway. I'll have a look at refactoring it.

Thank you very much for discovering this.
